### ISSUE TITLE
DateTime string formats /  ISO 8601

### ISSUE THREAD

**Issue by jwbrew** [biocad/openapi3#16]
I'm using Servant & OpenAPI to generate a schema for code generation via [openapi-generator](https://github.com/OpenAPITools/openapi-generator), targeting Dart as a client language. 

The current implmentations of `declareNamedSchema` for `UTCTime`, and `toParamSchema` for `UCTime` and `ZonedTime` generate a schema of `type: string` with a custom `format` as a specific representation of each type. This breaks codegen in the Dart client, as it defines the required type as a `String` where a `DateTime` would be more appropriate. This seems to be down to the `format` being a custom representation, rather than one of openapi3's built-in format values.

Given the custom formats as currently returned (`yyyy-mm-ddThh:MM:ssZ` and `yyyy-mm-ddThh:MM:ss+hhMM`) are (as far as I can see) [ISO8601](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14) - and therefore [openapi3 date-time](https://swagger.io/docs/specification/data-models/data-types/) compliant, I would like to open a PR that changes the implementation `declareNamedSchema` for `UTCTime`, and `toParamSchema` for `UTCTime` and `ZonedTime` to return `format: "date-time"`. 

Is this a reasonable thing to do?

As ever with matters involving dates and times the devil is hiding in every detail and I expect there's a subtle detail I've missed here. Hopefully not however, as these changes help avoid wrapping a whole load of types to generate the right schema!

**Comment by maksbotan** [comment]
Hi @jwbrew,

My guess is that UTCTime uses this format to specifically indicate that the time zone is always `Z`. If we switch to `format: date-time`, the reader of the spec will not know this detail. What do you think?

As for `ToParamSchema` instance for `ZonedTime`, I suppose this is a mistake, I will fix that.

**Comment by maksbotan** [comment]
I've changed `ToParamSchema ZonedTime` instance for now. Let me know what you think about my comment above.

**Comment by jwbrew** [comment]
> My guess is that UTCTime uses this format to specifically indicate that the time zone is always Z. If we switch to format: date-time, the reader of the spec will not know this detail. What do you think?

Yup I think that makes complete sense. IIRC I remember thinking something similar some time after posting the original comment. Thanks for making the change to `ZonedTime`, it's going to make things a whole lot more straightforward now!

**Comment by maksbotan** [comment]
You are welcome!

I'll keep this issue open for some time, to think about it once again later.

**Comment by maksbotan** [comment]
Well, that makes sense. After all, always-`Z` time is certainly a subtype of ISO8601 time. Can you make a PR for that change?

BTW, did you consider using UNIX timestamps instead of strings? IMO that representation is far more convenient.

**Comment by dhess** [comment]
Assuming the PR is as simple as I'm imagining, I'll try to find the time soon, yes.

I didn't consider UNIX timestamps. We want to store YYYY-MM-DD and HH:MM:SS uniformly on the server side as UTC and then use the client's browser to render those dates+times as appropriate for the client's locale. There's a straightforward way (or should be, per my argument above) to go from `UTCTime` in Haskell to `date-time` in OpenAPI, and then to `Date` in JavaScript, so what could be more convenient than that for our purposes? :) 

I'm all ears though if you think there's a better/simpler way to achieve this with UNIX timestamps.
